dot/rpc/subscription/listeners.go

```diff
@@ -285,4 +296,67 @@ func (l *RuntimeVersionListener) Listen() {
 
 // Stop to runtimeVersionListener not implemented yet because the listener
 // does not need to be stoped
-func (l *RuntimeVersionListener) Stop() {}
+func (l *RuntimeVersionListener) Stop() error { return nil }
+
+// GrandpaJustificationListener struct has the finalisedCh and the context to stop the goroutines
+type GrandpaJustificationListener struct {
+	cancel        chan interface{}
+	cancelTimeout time.Duration
+	done          chan interface{}
+	wsconn        *WSConn
+	subID         uint32
+	finalisedChID byte
+	finalisedCh   chan *types.FinalisationInfo
+}
+
+const grandpaJustifications = "grandpa_justifications"
+
+// Listen will start goroutines that listen to the finaised blocks
+func (g *GrandpaJustificationListener) Listen() {
+	// listen for finalised headers
+	go func() {
+		defer func() {
+			close(g.done)
+			close(g.finalisedCh)
+		}()
```

Comment:
you need to unsubscribe from the subscription with `blockState.UnregisterFinalisedChannel(finalisedChID)`, don't close the channel here. in general channels should be closed by the writer, otherwise the writer might be writing to a closed channel (#1701)